resources/BUILD.gn

```diff
@@ -0,0 +1,163 @@
+    deps += [ "//ui/views/resources" ]
+  }
+}
+
+repack("xwalk_resources_100_percent_pak") {
+  output = "$root_out_dir/xwalk_100_percent.pak"
+  sources = [
+    "$root_gen_dir/blink/public/resources/blink_image_resources_100_percent.pak",
+    "$root_gen_dir/components/components_resources_100_percent.pak",
+    "$root_gen_dir/content/app/resources/content_resources_100_percent.pak",
+    "$root_gen_dir/ui/resources/ui_resources_100_percent.pak",
+  ]
+  deps = [
+    "//components/resources",
+    "//content/app/resources:resources_grit",
+    "//third_party/WebKit/public:image_resources_grit",
+    "//ui/resources",
+  ]
+  if (toolkit_views) {
+    deps += [ "//ui/views/resources:resources_grd_grit" ]
+    sources +=
+        [ "$root_gen_dir/ui/views/resources/views_resources_100_percent.pak" ]
+  }
+}
+
+repack("xwalk_resources_200_percent_pak") {
+  output = "$root_out_dir/xwalk_200_percent.pak"
+  sources = [
+    "$root_gen_dir/blink/public/resources/blink_image_resources_200_percent.pak",
+    "$root_gen_dir/components/components_resources_200_percent.pak",
+    "$root_gen_dir/content/app/resources/content_resources_200_percent.pak",
+    "$root_gen_dir/ui/resources/ui_resources_200_percent.pak",
+  ]
+  deps = [
+    "//components/resources",
+    "//content/app/resources:resources_grit",
+    "//third_party/WebKit/public:image_resources_grit",
+    "//ui/resources",
+  ]
+  if (toolkit_views) {
+    sources +=
+        [ "$root_gen_dir/ui/views/resources/views_resources_200_percent.pak" ]
+    deps += [ "//ui/views/resources:resources_grd_grit" ]
+  }
+}
+
+repack("xwalk_resources_300_percent_pak") {
+  output = "$root_out_dir/xwalk_300_percent.pak"
+  sources = [
+    "$root_gen_dir/components/components_resources_300_percent.pak",
+    "$root_gen_dir/ui/resources/ui_resources_300_percent.pak",
+  ]
+  deps = [
+    "//components/resources",
+    "//ui/resources",
+  ]
+  if (toolkit_views) {
+    sources +=
+        [ "$root_gen_dir/ui/views/resources/views_resources_300_percent.pak" ]
+    deps += [ "//ui/views/resources:resources_grd_grit" ]
+  }
+}
+
+grit("generate_xwalk_resources") {
+  outputs = [
+    "$root_gen_dir/xwalk/resources/grit/xwalk_resources.h",
+    "$root_gen_dir/xwalk/resources/grit/xwalk_resources_map.cc",
+    "$root_gen_dir/xwalk/resources/grit/xwalk_resources_map.h",
+    "$root_gen_dir/xwalk/resources/xwalk_resources.pak",
+    "$root_gen_dir/xwalk/resources/xwalk_resources.rc",
+  ]
+  source = "//xwalk/runtime/resources/xwalk_resources.grd"
+  resource_ids = "//xwalk/resources/resource_ids"
+}
+
+copy("xwalk_resources") {
+  outputs = [
+    "$root_gen_dir/xwalk/grit/{{source_file_part}}",
+  ]
+  sources = [
+    "$root_gen_dir/xwalk/resources/grit/xwalk_resources.h",
+    "$root_gen_dir/xwalk/resources/grit/xwalk_resources_map.cc",
+    "$root_gen_dir/xwalk/resources/grit/xwalk_resources_map.h",
+  ]
+  deps = [
+    ":generate_xwalk_resources",
+  ]
+}
+
+grit("generate_xwalk_strings") {
```

Comment:
by the way, I will remove line43, the "$root_gen_dir/xwalk/locales/en-US.pak" should not be packed into xwalk.pak.